JsonLogic/JsonWriterExtensions.cs

```diff
@@ -25,8 +25,7 @@ public static void WriteRule(this Utf8JsonWriter writer, Rule? rule, JsonSeriali
 		}
 
 #pragma warning disable IL2026, IL3050
-		// using the options.Write() extension here for some reason causes a StackOverflow
-		JsonSerializer.Serialize(writer, rule, rule.GetType(), LogicSerializerContext.OptionsManager.SerializerOptions);
+		options.Write(writer, rule, rule.GetType());	
```

Comment:
FYI still need to use Suppression attributes instead of `#pragma warning disable` because library consumers don't see the `#pragma warning` and the callers will get AOT warnings on our code. The AOT analyzer only sees attributes so we need to suppress that way. See #636

Reply:
Added it in #638